dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/IntrinsicsInlining.java

```diff
@@ -0,0 +1,117 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.exception.ParsingException;
+import com.dat3m.dartagnan.expression.BNonDet;
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.expression.ExpressionFactory;
+import com.dat3m.dartagnan.expression.INonDet;
+import com.dat3m.dartagnan.expression.type.BooleanType;
+import com.dat3m.dartagnan.expression.type.IntegerType;
+import com.dat3m.dartagnan.expression.type.TypeFactory;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.Thread;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.core.Event;
+import com.dat3m.dartagnan.program.event.functions.DirectFunctionCall;
+import com.dat3m.dartagnan.program.event.functions.DirectValueFunctionCall;
+import com.google.common.primitives.UnsignedInteger;
+import com.google.common.primitives.UnsignedLong;
+import org.sosy_lab.common.configuration.Configuration;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class IntrinsicsInlining implements ProgramProcessor {
+
+    private int constantId;
+
+    private IntrinsicsInlining() {
+    }
+
+    public static IntrinsicsInlining fromConfig(Configuration ignored) throws InvalidConfigurationException {
+        return new IntrinsicsInlining();
+    }
+
+    @Override
+    public void run(Program program) {
+        constantId = 0;
+        for (Thread thread : program.getThreads()) {
+            run(thread);
+        }
+    }
+
+    private void run(Thread thread) {
+        for (Event event : List.copyOf(thread.getEvents())) {
+            if (!(event instanceof DirectFunctionCall call)) {
```

Comment:
Can we simply call `getEvents()` with parameter `DirectFunctionCall` to only get instances of that class?

Reply:
Btw. `thread.getEvents()` returns a fresh copy, so List.copyOf is not needed.